transactions/transactor.go

```diff
@@ -334,7 +334,7 @@ func (t *Transactor) HashTransaction(args SendTxArgs) (validatedArgs SendTxArgs,
 	newNonce := hexutil.Uint64(nonce)
 	newGas := hexutil.Uint64(gas)
 	validatedArgs.Nonce = &newNonce
-	if args.GasPrice != nil {
+	if args.IsDynamicFeeTx() {
```

Comment:
```suggestion
	if !args.IsDynamicFeeTx() {
```
I think we want to use the `MaxFeePerGas` and `MaxPriorityFeePerGas` in case they are available